app/src/main/java/de/danoeh/antennapod/fragment/FeedInfoFragment.java

```diff
@@ -233,6 +242,30 @@ private void showFeed() {
         }
 
         txtvUrl.setText(feed.getDownload_url() + " {fa-paperclip}");
+
+        if (feed.getPaymentLinks() == null || feed.getPaymentLinks().size() == 0) {
+            lblSupport.setVisibility(View.GONE);
+            txtvFundingUrl.setVisibility(View.GONE);
+        } else {
+            lblSupport.setVisibility(View.VISIBLE);
+            ArrayList<FeedFunding> fundingList = feed.getPaymentLinks();
+            String str = "";
+            ArrayList<String> seen = new ArrayList<String>();
```

Comment:
I think I would use a `HashSet` here. The order of the list is not important, so a set is more fitting conceptually